File: repair-management-api/Models/Device.cs
using RepairManagementApi.Enums;

namespace RepairManagementApi.Models;

public class Device
{
    public Guid Id { get; set; }
    public Guid CustomerId { get; set; }
    public Customer? Customer { get; set; }
    public Guid BranchId { get; set; }
    public Branch? Branch { get; set; }
    public string Brand { get; set; } = string.Empty;
    public string Model { get; set; } = string.Empty;
    public string? SerialNumber { get; set; }
    public DeviceType DeviceType { get; set; }
    public DateTime CreatedAtUtc { get; set; }
    public DateTime UpdatedAtUtc { get; set; }
}

File: repair-management-app/src/types/device.ts
export type DeviceType = "Mobile" | "Laptop" | "Tablet" | "Desktop" | "Other";

export type DeviceListItem = {
  id: string;
  customerId: string;
  brand: string;
  model: string;
  imeiOrSerialNumber: string | null;
  deviceType: DeviceType;
  createdAtUtc: string;
};

export type DeviceResponse = {
  id: string;
  customerId: string;
  branchId: string;
  brand: string;
  model: string;
  imeiOrSerialNumber: string | null;
  deviceType: DeviceType;
  createdAtUtc: string;
  updatedAtUtc: string;
};

export type CreateDeviceRequest = {
  customerId: string;
  branchId: string;
  brand: string;
  model: string;
  imeiOrSerialNumber: string | null;
  deviceType: DeviceType;
};

export type UpdateDeviceRequest = {
  brand: string;
  model: string;
  imeiOrSerialNumber: string | null;
  deviceType: DeviceType;
};

export type DeviceLookupResponse = {
  deviceId: string;
  customerId: string;
  customerName: string;
  customerPhone: string;
  brand: string;
  model: string;
  serialNumber: string | null;
  deviceType: DeviceType;
};
